### PR TITLE
Extend time range picker

### PR DESCRIPTION
**Pull request by raczyk**
# What

- Extend time range picker
- Screenshots
Screenshot 2022-11-02 at 13 55 13

## Testing

- [ ] Is this change covered by the unit tests?

- [ ] Is this change covered by the integration tests?

- [ ] Is this change covered by the automated acceptance tests? (if applicable)

## Compatibility

- [x] Does this change maintain backward compatibility?

### CODE REVIEW

File: src/components/timeRangePicker/TimeRangePicker.tsx [hunk 1]
```diff
@@ -118,6 +163,21 @@ export function TimeRangePicker({
     }
   };
 
+  // Function to update date range when arrows are clicked
+  const updateCurrentTime = (direction: string) => {
```
Comment (nowyDEV) [inline src/components/timeRangePicker/TimeRangePicker.tsx:167]:
```suggestion
  const updateCurrentTime = (direction: "forward" | "backward") => {
```

File: src/components/timeRangePicker/TimeRangePicker.tsx [hunk 2]
```diff
@@ -130,6 +190,61 @@ export function TimeRangePicker({
     onChange({ start: timeStart, end: timeEnd });
   };
 
+  if (type === "Days") {
+    return (
+      <>
+        <FlexContainer alignContent="center">
+          <FlexContainer grow={1} style={{ marginRight: "1rem" }}>
```
Comment (nowyDEV) [inline src/components/timeRangePicker/TimeRangePicker.tsx:197]: Let's not use inline React styles, we have other tools to do this.

File: src/components/timeRangePicker/TimeRangePicker.tsx [hunk 3]
```diff
@@ -295,3 +410,44 @@ const getDateWithTime = (date: Date, time: string): Date => {
...
+      dateFormat
+    )}`;
+  }
+  if (rangeType === RANGE_TYPE.YEAR) {
+    return `${format(new Date(date), dateFormat)} - ${format(
+      addYears(date, 1),
+      dateFormat
+    )}`;
+  }
+  return format(new Date(date), dateFormat);
```
Comment (nowyDEV) [inline src/components/timeRangePicker/TimeRangePicker.tsx:433]: I think it's more readable this way

```suggestion
  if (rangeType === RANGE_TYPE.YEAR) {
    return `${format(new Date(date), dateFormat)} - ${format(
      addYears(date, 1),
      dateFormat
    )}`;
  }
  
   if (rangeType === RANGE_TYPE.DAY) {
      return format(new Date(date), dateFormat);
   }
```

File: src/components/timeRangePicker/TimeRangePicker.tsx [hunk 4]
```diff
@@ -130,6 +190,61 @@ export function TimeRangePicker({
     onChange({ start: timeStart, end: timeEnd });
   };
 
+  if (type === "Days") {
```
Comment (mrMarshalX) [inline src/components/timeRangePicker/TimeRangePicker.tsx:193]:
```suggestion
  if (type === RANGE_TYPE.DAY) {
```

**Comment by nowyDEV** [comment]
There is a console error inside preview, something is wrong with the story file.

Screenshot_2022-11-03_09-32-32

File: src/components/timeRangePicker/TimeRangePicker.tsx (hunk 2 above)
Comment (mrMarshalX) [inline src/components/timeRangePicker/TimeRangePicker.tsx:197]:
```suggestion
          <FlexContainer grow={1} style={{ marginRight: theme.spacing.spacing04 }}>
```

File: src/components/timeRangePicker/TimeRangePicker.tsx (hunk 3 above)
Comment (mrMarshalX) [inline src/components/timeRangePicker/TimeRangePicker.tsx:433]:
```suggestion
  }

  return format(new Date(date), dateFormat);
```

File: src/components/timeRangePicker/TimeRangePicker.tsx (hunk 4 above)
Comment (raczyk) [reply src/components/timeRangePicker/TimeRangePicker.tsx:193]: RANGE_TYP is a different enum, but I've created another one.

**Review by mrMarshalX — DISMISSED** [review]

File: src/components/timeRangePicker/TimeRangePicker.tsx [hunk 5]
```diff
@@ -130,6 +192,61 @@ export function TimeRangePicker({
     onChange({ start: timeStart, end: timeEnd });
   };
 
+  if (type === TIME_TYPE.DAYS) {
+    return (
+      <>
```
[the remark this answers is not in the document]
Comment (raczyk) [reply src/components/timeRangePicker/TimeRangePicker.tsx]: removed :)

**Review by nowyDEV — APPROVED** [review]

**Review by Bosmanfrx — APPROVED** [review]

**Review by janosio-nordcloud — APPROVED** [review]

**Pull request merged**